src/error.rs

```diff
@@ -127,6 +134,12 @@ pub enum Error {
     /// Trailing data was found while parsing DER-encoded input for the named type.
     TrailingData(DerTypeId),
 
+    /// The certificate is not valid for the name it is being validated for.
+    ///
+    /// Variant without context for use in no-`alloc` environments. See `UnexpectedCertName` for
```

Comment:
I think this is still referring to the new name for `CertNotValidForName` that you backed out.

But more interestingly, now that `InvalidNameContext` handles the alloc vs not conditions isn't `UnexpectedCertNameSimple` redundant?

Reply:
Good catch, cleaned it out.